requirements.txt

```diff
@@ -1,23 +1,23 @@
 # Python dependencies required for development
-astunparse
-cmake
-expecttest>=0.3.0
-filelock
-fsspec
-hypothesis
+astunparse==1.6.3
+cmake==4.0.3
```

Comment:
This broke downstream builds in TheRock: https://github.com/ROCm/TheRock/actions/runs/16787789191/job/47543320786#step:11:2286, where we install this requirements file with `cmake<4`: https://github.com/ROCm/TheRock/blob/377eeb2d50364c943b3745a1c98931e0e984d15e/external-builds/pytorch/build_prod_wheels.py#L608-L618

We can adapt our downstream code, but I'm wondering
1. Why pin these versions on the release branch in this pytorch repository itself, instead of via some downstream build process?
2. Why choose CMake >= 4? That is still unstable and may not be supported in other ROCm projects: https://github.com/ROCm/TheRock/issues/318

Reply:
Removed the conflicting pin from downstream: https://github.com/ROCm/TheRock/pull/1212